src/dynamodb/gameStateDBService.ts

```diff
@@ -0,0 +1,127 @@
+import { DocumentClient } from 'aws-sdk/clients/dynamodb';
+import { GAME_STATE_TABLE } from '../utils/constants';
+import { HongKongWall } from '../games/mahjong/Wall/version/HongKongWall';
+import { DB } from './db';
+import { GameState, UserHand } from '../models/GameState';
+import { getHandByConnectionId, parseDynamoDBAttribute, parseDynamoDBItem } from './dbHelper';
+
+/**
+ * Initialize the game by generating a mahjong wall,
+ * 4 hands of mahjong, and save the initial game state to the db
+ * @param {string} gameId game Id
+ * @param {string} connectionIds connection Ids of all the users in a game
+ */
+export const initGameState = async (gameId: string, connectionIds: string[]): Promise<GameState> => {
+  const initialWall = new HongKongWall();
+
+  // Generate hand for each user
+  const hands: UserHand[] = [];
+  connectionIds.forEach((connectionId: string) => {
+    const hand = {
+      connectionId,
+      hand: initialWall.generateHand(),
+    };
+    hands.push(hand);
+  });
+
+  const initialGame: GameState = {
+    gameId,
+    wall: initialWall.getTiles(), // array of tiles
+    hands, // current hands of users TODO: can remove this attribute if not needed
+    currentIndex: 13 * 4, // index of the tile array after game init
```

Comment:
Is this 13 * 4 - 1 since 0 is accounted for? We could also get initial currentIndex from wall.getCurrentTileIndex()

Reply:
the currentIndex means the next available tile's index, so if we used 13 * 4 = 52 tiles (at index 51) after game init the current index or the next available tile's index would be 52 (51 + 1, index 51 tile is already used)

Reply:
I see. Makes sense